shelley-ma/shelley-ma-test/src/Test/Cardano/Ledger/ShelleyMA/Timelocks.hs

```diff
@@ -1,74 +1,111 @@
 {-# LANGUAGE DataKinds #-}
 {-# LANGUAGE FlexibleContexts #-}
 {-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+{-# OPTIONS_GHC -Wno-orphans #-} -- orphan Arbitrary instances
+
 
 module Test.Cardano.Ledger.ShelleyMA.Timelocks
   ( timelockTests,
-    testT,
+    sizedTimelock,
+    MultiSig,
   )
 where
 
-import Cardano.Binary
-  ( Annotator (..),
-    FromCBOR (fromCBOR),
-    FullByteString (Full),
-    ToCBOR (toCBOR),
-  )
+import Data.Coders(roundTripAnn, embedTripAnn)
+import Shelley.Spec.Ledger.Keys (KeyHash (..))
+import Cardano.Ledger.Era
+import Cardano.Binary( Annotator (..), FromCBOR(..), ToCBOR(..) )
 import Cardano.Ledger.ShelleyMA.Timelocks
 import Cardano.Slotting.Slot (SlotNo (..))
-import Codec.CBOR.Read (DeserialiseFailure, deserialiseFromBytes)
-import Codec.CBOR.Write (toLazyByteString)
 import qualified Data.ByteString.Lazy as Lazy
 import Data.Sequence.Strict (fromList)
 import Shelley.Spec.Ledger.BaseTypes (StrictMaybe (SJust, SNothing))
 import Test.Cardano.Ledger.ShelleyMA.TxBody (TestEra)
 import Test.Tasty
 import Test.Tasty.QuickCheck hiding (scale)
+import Shelley.Spec.Ledger.Scripts(MultiSig)
+import Test.Shelley.Spec.Ledger.Serialisation.Generators(mkDummyHash) -- imports arbitray instance for MultiSig
```

Comment:
In the Shelley tests, we're careful to distinguish between the serialisation tests and other tests. This is because the generator strategy is so different. It looks like this module purely consists of serialisation tests, so it would be better to make that clear in the module path. (E.g. in Shelley we have `Test.Shelley.Spec.Ledger.Serialisation.Tripping.CBOR`)

Reply:
That is a good idea, I should probably put the Coders tests in there as well.